### PR TITLE
ARROW-1885: [Java] Restore MapVector class names prior to ARROW-1710

### PR DESCRIPTION
**Pull request by BryanCutler**
Changes to MapVector were not part of the intended goal for ARROW-1710 and should be restored to `NullableMapVector` -> `MapVector` -> `AbstractMapVector` as existed prior to #1341.

### CODE REVIEW

**Comment by BryanCutler** [comment]
@siddharthteotia I manually checked differences of this change to those done in #1341 to ensure readers/writers and others are back to the previous state, but please double check and let me know if you feel comfortable with this. 
cc @icexelloss @jacques-n

File: java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java [hunk 1]
```diff
@@ -21,480 +21,322 @@
...
-  public MapVector(String name, BufferAllocator allocator, FieldType fieldType, CallBack callBack) {
-    super(name, checkNotNull(allocator), fieldType, callBack);
-    this.validityBuffer = allocator.getEmpty();
-    this.validityAllocationSizeInBytes = BitVectorHelper.getValidityBufferSize(BaseValueVector.INITIAL_VALUE_ALLOCATION);
+  @Override
+  public FieldReader getReader() {
+    return reader;
+  }
+
+  transient private MapTransferPair ephPair;
```
Comment (icexelloss) [inline java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java]: `ephPair` seems wrong

**Comment by icexelloss** [comment]
LGTM except for what seems to be a typo .

File: java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java (hunk 1 above)
Comment (BryanCutler) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java]: how so, you mean the name?

Comment (BryanCutler) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java]: keep in mind the diff is showing the result of renaming `NonNullableMapVector` to `MapVector` and none of the code was actually changed, just the class name

Comment (icexelloss) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java]: Oh sorry I should be clear - the variable name `ephPair` seems wrong - I don't know if it has always been `ephPair` or somehow changed during refactoring. Either way, seems worth fixing.

Comment (BryanCutler) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java]: Oh ok, I'm not sure of the reason for the name, this is how it was previously to ARROW-1710.  I'd prefer to keep the changes here to that scope so we can be sure they are restored correctly.

**Comment by siddharthteotia** [comment]
+1

File: java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java (hunk 1 above)
Comment (icexelloss) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java]: Ok, sounds good.

**Review by icexelloss — APPROVED** [review]

**Comment by BryanCutler** [comment]
Thanks all for the quick reviews!

**Pull request closed without merging**